flocker/control/_protocol.py

```diff
@@ -354,15 +355,19 @@ def connectionLost(self, reason):
         self._pinger.stop()
 
 
-DEPLOYMENT_CONFIG = Field(u"configuration", repr,
+# These two logging fields use wire_encode as the serializer so that they can
+# share the encoding cache with the network code related to this logging.  This
+# reduces the overhead of logging these (potentially quite large) data
+# structures.
+DEPLOYMENT_CONFIG = Field(u"configuration", wire_encode,
                           u"The cluster configuration")
-CLUSTER_STATE = Field(u"state", repr,
+CLUSTER_STATE = Field(u"state", wire_encode,
```

Comment:
I think you're right.  Using `wire_encode` directly here is bypassing the caching I was trying to bring in.